File: semestr-8/Jezyk C#/alexander/Csharp/Archiwum/2020/ID06IO2/BuilderAndFabric/BuilderAndFabric/AnimalFabric.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BuilderAndFabric
{
    delegate Animal MakeAnimal();
    class AnimalFabric
    {
        private static Dictionary<string, MakeAnimal> map = MakeMap();
        private static Dictionary<string, MakeAnimal> MakeMap()
        {
            Dictionary<string, MakeAnimal> map = new Dictionary<string, MakeAnimal>();
            map.Add("Cat", () => new Cat());
            map.Add("Fly", () => new Fly());
            return map;
        }
        private static Animal mk() { return new Cat(); }
        public static Animal make(string kind) { return map[kind](); }
        public static List<string> keys()
        {
            List<string> lst = map.Keys.ToList();
            lst.Sort();
            return lst;
        }
    };


    class Animal
    {

    }
    class Cat : Animal
    {

    }
    class Fly : Animal
    {

    }
}

File: semestr-8/Programowanie obiektowe w Java/alexander/POJ/Archiwum/2020w/ID07IO1AB/FabricAnimals/src/AnimalFabric.java

import java.util.List;
import java.util.HashMap;
import java.util.stream.Collectors;

public class AnimalFabric
{
   private static final HashMap<String,AnimalMaker> map=makemap();
   private static HashMap<String,AnimalMaker> makemap()
   {
      final HashMap<String,AnimalMaker> map=new HashMap<String,AnimalMaker>();
      map.put("Cat",(name) -> new Cat(name));
      map.put("Dog",(name) -> new Dog(name));
      map.put("Fly",(name) -> new Fly(name));
      return map;
   }
   public static Animal make(String kind,String name)
   {
      return map.get(kind).make(name);
   }
   public static List<String> keys()
   {
      return map.keySet().stream().sorted().collect(Collectors.toList());
   }
}
